client_factory.go

```diff
@@ -46,6 +46,7 @@ func NewHttpAuthClient(
 		"The number of failed authentications.",
 	)
 	return &httpAuthClient{
+		config:                config,
```

Comment:
I think it would be worthwhile to return `PubKey` and` Password` separately instead of returning the full config.